app/src/main/java/org/schabi/newpipe/info_list/InfoItemDialog.java

```diff
@@ -1,54 +1,86 @@
 package org.schabi.newpipe.info_list;
 
-import android.app.Activity;
 import android.app.AlertDialog;
-import android.content.DialogInterface;
+import android.app.Dialog;
+import android.content.DialogInterface.OnClickListener;
+import android.os.Bundle;
 import android.view.View;
 import android.widget.TextView;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.FragmentManager;
 
 import org.schabi.newpipe.R;
-import org.schabi.newpipe.extractor.stream.StreamInfoItem;
 
-public class InfoItemDialog {
-    private final AlertDialog dialog;
+import java.io.Serializable;
 
-    public InfoItemDialog(@NonNull final Activity activity,
-                          @NonNull final StreamInfoItem info,
+public class InfoItemDialog  {
+
+    private final ItemDialogFragment idf;
+
+    public InfoItemDialog(@NonNull final String title,
+                          final String additionalInfo,
                           @NonNull final String[] commands,
-                          @NonNull final DialogInterface.OnClickListener actions) {
-        this(activity, commands, actions, info.getName(), info.getUploaderName());
+                          @NonNull final OnClickListener actions) {
```

Comment:
These are the first steps towards consolidating how some of these popups are created. 
The current implementation works for items in regular streams, channels and subscriptions.